pkg/devicetwin/dtcontext/dtcontext_test.go

```diff
@@ -0,0 +1,493 @@
+			name:     "KnownDevice",
+			deviceID: "DeviceB",
+			want:     &dttype.Device{},
+			wantBool: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, gotBool := dtc.GetDevice(test.deviceID)
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("DTContext.GetDevice() got = %v, want %v", got, test.want)
+				return
+			}
+			if gotBool != test.wantBool {
+				t.Errorf("DTContext.GetDevice() gotBool = %v, want %v", gotBool, test.wantBool)
+				return
+			}
+		})
+	}
+}
+
+//Function TestSend is function to test Send().
+func TestSend(t *testing.T) {
+	payload := dttype.MembershipUpdate{AddDevices: []dttype.Device{{ID: "DeviceA", Name: "Router", State: "unknown"}}}
+	content, err := json.Marshal(payload)
+	if err != nil {
+		t.Errorf("Got error on marshalling: %v", err)
+	}
+	commChan := make(map[string]chan interface{})
+	receiveCh := make(chan interface{}, 1)
+	commChan[dtcommon.TwinModule] = receiveCh
+	var msg = &model.Message{
+		Content: content,
+	}
+	dtc := &DTContext{
+		CommChan: commChan,
+	}
+	tests := []struct {
+		name     string
+		identity string
+		action   string
+		module   string
+		msg      *model.Message
+		wantErr  bool
+	}{
+		{
+			//Failure Case
+			name:     "UnknownModule",
+			identity: "",
+			action:   dtcommon.SendToCloud,
+			module:   dtcommon.CommModule,
+			msg:      msg,
+			wantErr:  true,
+		},
+		{
+			//Success Case
+			name:     "KnownModule",
+			identity: "",
+			action:   dtcommon.SendToCloud,
+			module:   dtcommon.TwinModule,
+			msg:      msg,
+			wantErr:  false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if err := dtc.Send(test.identity, test.action, test.module, test.msg); (err != nil) != test.wantErr {
+				t.Errorf("DTContext.Send() error = %v, wantErr %v", err, test.wantErr)
+			}
+		})
+	}
+}
+
+//TestBuildModelMessage is to test BuildModelMessage().
+func TestBuildModelMessage(t *testing.T) {
+	dtc := &DTContext{}
+	payload := dttype.MembershipUpdate{AddDevices: []dttype.Device{{ID: "DeviceA", Name: "Router", State: "unknown"}}}
+	content, err := json.Marshal(payload)
+	if err != nil {
+		t.Errorf("Error on Marshalling: %v", err)
+	}
+	tests := []struct {
+		name      string
+		group     string
+		parentID  string
+		resource  string
+		operation string
+		content   interface{}
+		want      *model.Message
+	}{
+		{
+			name:      "BuildModelMessageTest",
+			group:     "resource",
+			parentID:  "",
+			resource:  "membership/detail",
+			operation: "get",
+			content:   content,
+			want:      &model.Message{},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got := dtc.BuildModelMessage(test.group, test.parentID, test.resource, test.operation, test.content)
+			if reflect.TypeOf(got) != reflect.TypeOf(test.want) {
+				t.Errorf("DTContext.BuildModelMessage() = %v, want %v", got, test.want)
+				return
+			}
+			if got.Header.ParentID != test.parentID {
```

Comment:
Can we fill want in test cases and have 1 condition reflect.DeepEqual to verify got and want ?